### PR TITLE
Start section on commonly encounted unsupported parfors code.

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title. This covers a couple of commonly reported problems.

### CODE REVIEW

File: docs/source/user/parallel.rst [hunk 1]
```diff
@@ -224,6 +224,41 @@ give an equivalence parallel implementation using :func:`~numba.guvectorize`,
 it would require a pervasive change that rewrites the code to extract kernel
 computation that can be parallelized, which was both tedious and challenging.
 
+Unsupported Operations
+======================
+
+This section contains a non-exhaustive list of commonly encountered but
+currently unsupported features:
+
+#. Concurrent write operations on container types (i.e. lists, sets and
```
Comment (sklam) [inline docs/source/user/parallel.rst:233]: Should we give it a title for quick look up? e.g. "List operations are not thread safe."

File: docs/source/user/parallel.rst [hunk 2]
```diff
@@ -224,6 +224,41 @@ give an equivalence parallel implementation using :func:`~numba.guvectorize`,
...
+        z = []
+        for i in prange(10000):
+            z.append(i)
+        return z
+
+   It is highly likely that the above will result in corruption or an access
+   violation as containers require thread-safety under mutation but this feature
+   is not implemented.
+
+#. The use of the induction variable induced by a ``prange`` based loop in
```
Comment (sklam) [inline docs/source/user/parallel.rst:247]: maybe title: "Induction variable is not associated with thread id."

**Review by sklam — COMMENTED** [review]
Thanks for the patch! I think adding a title sentence for each example can allow users to read it faster.

File: docs/source/user/parallel.rst (hunk 1 above)
Comment (stuartarchibald) [reply docs/source/user/parallel.rst]: Good idea, added similar titles in 57b0028

**Review by sklam — APPROVED** [review]
LGTM

**Comment by stuartarchibald** [comment]
CI passed. Thanks for the review @sklam.

**Pull request merged**